crates/ruff_linter/src/rules/flake8_return/rules/function.rs

```diff
@@ -614,17 +613,17 @@ fn unnecessary_assign(checker: &Checker, stack: &Stack) {
             let delete_return =
                 edits::delete_stmt(stmt, None, checker.locator(), checker.indexer());
 
-            // Replace the `x = 1` statement with `return 1`.
-            let content = checker.locator().slice(assign);
-            let equals_index = content
-                .find('=')
-                .ok_or(anyhow::anyhow!("expected '=' in assignment statement"))?;
-            let after_equals = equals_index + 1;
+            let eq_token = checker
+                .tokens()
+                .iter()
```

Comment:
Iterating over all tokens is unnecessarily slow. Especially given that you know exactly what token you're looking for. You can use `checker.tokens().in_range(assign.range()`  or `after(assign.start)` instead